src/ray/core_worker/object_interface.cc

```diff
@@ -1,17 +1,20 @@
 #include "ray/core_worker/object_interface.h"
 #include "ray/common/ray_config.h"
+#include "ray/core_worker/store_provider/local_plasma_provider.h"
 #include "ray/core_worker/store_provider/plasma_store_provider.h"
 
 namespace ray {
 
 CoreWorkerObjectInterface::CoreWorkerObjectInterface(
     WorkerContext &worker_context, std::unique_ptr<RayletClient> &raylet_client,
     const std::string &store_socket)
-    : worker_context_(worker_context), raylet_client_(raylet_client) {
-  store_providers_.emplace(
-      static_cast<int>(StoreProviderType::PLASMA),
-      std::unique_ptr<CoreWorkerStoreProvider>(
-          new CoreWorkerPlasmaStoreProvider(store_socket, raylet_client_)));
+    : worker_context_(worker_context),
+      raylet_client_(raylet_client),
+      store_socket_(store_socket) {
+  store_providers_.emplace(static_cast<int>(StoreProviderType::LOCAL_PLASMA),
```

Comment:
Nit. Can we change `store_providers_`'s key type to `StoreProviderType`? That would be a bit clearer. And we don't have to cast the type everywhere. 
We can use an alias if the type becomes too long, https://github.com/ray-project/ray/blob/master/src/ray/raylet/worker_pool.h#L19

Reply:
Thanks. This does make it more readable:)

Reply:
I changed task submitters and receivers btw.